python/ray/tune/schedulers/pbt.py

```diff
@@ -628,27 +664,59 @@ def _checkpoint_or_exploit(
         if trial in upper_quantile:
             # The trial last result is only updated after the scheduler
             # callback. So, we override with the current result.
-            logger.debug("Trial {} is in upper quantile".format(trial))
-            logger.debug("Checkpointing {}".format(trial))
+            logger.debug(f"Trial {trial} is in upper quantile. Saving checkpoint.")
             if trial.status == Trial.PAUSED:
-                # Paused trial will always have an in-memory checkpoint.
-                state.last_checkpoint = trial.checkpoint
+                if trial.temporary_state.saving_to and isinstance(
+                    trial.temporary_state.saving_to, _FutureTrainingResult
+                ):
+                    logger.debug(f"Trial {trial} is still saving.")
+                    state.last_checkpoint = trial.temporary_state.saving_to
+                else:
+                    # Paused trial will always have an in-memory checkpoint.
+                    logger.debug(
+                        f"Trial {trial} is paused. Use last available "
+                        f"checkpoint {trial.checkpoint}."
+                    )
+                    state.last_checkpoint = trial.checkpoint
             else:
+                logger.debug(f"Instructing {trial} to save.")
                 state.last_checkpoint = tune_controller._schedule_trial_save(
-                    trial, CheckpointStorage.MEMORY, result=state.last_result
+                    trial, CheckpointStorage.PERSISTENT, result=state.last_result
```

Comment:
If we are PAUSED, I think `saving_to` should never be set -- we will have already processed the trial save, cleared `saving_to`, then set to PAUSED.

That leaves only the else case when we schedule a brand new save. In that case, let's do this future training result + resolve